### ISSUE TITLE
output resources cause unexpected change in 1.4.0 plan output 

### ISSUE THREAD

**Issue by paul-yolabs** [hashicorp/terraform#32815]
### Terraform Version

```shell
Terraform v1.4.0
on darwin_arm64
```

### Terraform Configuration Files

```
resource "null_resource" "thing" {
}

output "a_string" {
  value = "some-text"
}
```

### Debug Output

https://gist.github.com/paul-yolabs/5b487d82021f453da9f2f02ac2d97f29

### Expected Behavior

With terraform 1.3.9, after a "terraform apply", a "terraform plan" shows the expected "No changes" output.
```
terraform plan 
null_resource.thing: Refreshing state... [id=4090070604809349538]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are
needed.
```

### Actual Behavior

With 1.4.0, a "terraform plan" shows only the resource:
```
terraform plan   
null_resource.thing: Refreshing state... [id=4090070604809349538]
```

JSON plans are identical other than timestamps, "terraform" versions, and "ui": "1.0" vs "ui: "1.1"

### Steps to Reproduce

Use my sample code snippet
1. `terraform init`
2. `terraform apply`
3. `terraform plan`

The presence of the `output` resource shortcuts all of the plan output at 
https://github.com/hashicorp/terraform/blame/afcbff193bddf4b2611043a349a57915ce77b1c0/internal/command/jsonformat/plan.go#L87

While `len(changes) == 0` is true, `len(diffs.outputs) == 0` is false.  But the output diffs in this case (many cases, I'd assume) are `NoOp`, so this seems like it should have been handled before this when changes were.

I can create a PR, but there's a little bit of twistiness around knowing all the possible cases referenced in there, and I'm afraid I might miss something.

**Comment by jbardin** [comment]
Hi @paul-yolabs,

Thanks for filing the issue. This appears to be just a UI concern, as you said the plan is the same, and the process still exits with a detailed exit code of 0.

**Comment by paul-yolabs** [comment]
Yep, it's a UI concern not a planning bug, but the UI gives more info than the exit codes, so we use that output in our CI tooling.  Looks like tools like https://github.com/suzuki-shunsuke/tfcmt also do that.

Thanks for the quick fix, looking forward to 1.4.1.

**Comment by prowlaiii** [comment]
So, I'm unclear...
Is the lack of a summary message an issue or is that the new normal?
I certainly found it confusing and am going to drop back to 1.3.9.

**Comment by jbardin** [comment]
@prowlaiii, yes the missing output was restored via the linked PR, and the fix is planned to be released in v1.4.1.
